test/integration/gardenlet/networkpolicy/networkpolicy_suite_test.go

```diff
@@ -0,0 +1,187 @@
+	seedClusterIdentity = "seed"
+	syncPeriod          = 500 * time.Millisecond
+)
+
+var (
+	testRunID string
+
+	ctx = context.Background()
+	log logr.Logger
+
+	restConfig *rest.Config
+	testEnv    *gardenerenvtest.GardenerTestEnvironment
+	testClient client.Client
+	mgrClient  client.Reader
+
+	gardenNamespace *corev1.Namespace
+	seed            *gardencorev1beta1.Seed
+)
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(logger.MustNewZapLogger(logger.DebugLevel, logger.FormatJSON, zap.WriteTo(GinkgoWriter)))
+	log = logf.Log.WithName(testID)
+
+	By("starting test environment")
+	testEnv = &gardenerenvtest.GardenerTestEnvironment{
+		Environment: &envtest.Environment{},
+		GardenerAPIServer: &gardenerenvtest.GardenerAPIServer{
+			Args: []string{"--disable-admission-plugins=DeletionConfirmation,ResourceReferenceManager,ExtensionValidator"},
+		},
+	}
+
+	var err error
+	restConfig, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(restConfig).NotTo(BeNil())
+
+	DeferCleanup(func() {
+		By("stopping test environment")
+		Expect(testEnv.Stop()).To(Succeed())
+	})
+
+	By("creating testClient")
+	testClient, err = client.New(restConfig, client.Options{Scheme: kubernetes.GardenScheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	testRunID = utils.ComputeSHA256Hex([]byte(uuid.NewUUID()))[:8]
+	log.Info("Using test run ID for test", "testRunID", testRunID)
+
+	By("creating seed")
+	seed = &gardencorev1beta1.Seed{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "seed-",
+			Labels:       map[string]string{testID: testRunID},
+		},
+		Spec: gardencorev1beta1.SeedSpec{
+			Provider: gardencorev1beta1.SeedProvider{
+				Region: "region",
+				Type:   "providerType",
+			},
+			Networks: gardencorev1beta1.SeedNetworks{
+				Pods:     "10.0.0.0/16",
+				Services: "10.1.0.0/16",
+				Nodes:    pointer.String("10.2.0.0/16"),
+			},
+			DNS: gardencorev1beta1.SeedDNS{
+				IngressDomain: pointer.String("someingress.example.com"),
+			},
+		},
+	}
+	Expect(testClient.Create(ctx, seed)).To(Succeed())
+	log.Info("Created Seed for test", "seed", seed.Name)
+
+	patch := client.MergeFrom(seed.DeepCopy())
+	seed.Status.ClusterIdentity = pointer.String(seedClusterIdentity)
+	Expect(testClient.Status().Patch(ctx, seed, patch)).To(Succeed())
+
+	DeferCleanup(func() {
+		By("deleting seed")
+		Expect(testClient.Delete(ctx, seed)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("creating garden namespace for test")
+	gardenNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "garden-",
+		},
+	}
+
+	Expect(testClient.Create(ctx, gardenNamespace)).To(Succeed())
+	log.Info("Created Namespace for test", "namespaceName", gardenNamespace.Name)
+	testRunID = gardenNamespace.Name
+
+	DeferCleanup(func() {
+		By("deleting test namespace")
+		Expect(testClient.Delete(ctx, gardenNamespace)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("setup manager")
+	mgr, err := manager.New(restConfig, manager.Options{
```

Comment:
What about some object selectors here like in other tests?

Reply:
I added some for namespaces and the seed. We cannot influence labels for NetworkPolicy and Endpoints.